src/main/java/com/snowflake/kafka/connector/internal/streaming/DirectTopicPartitionChannel.java

```diff
@@ -640,9 +640,17 @@ public long fetchOffsetTokenWithRetry() {
    */
   private long streamingApiFallbackSupplier(
```

Comment:
i'm wondering if there is a risk of having race condition here - i.e. two callers invoking this method would invoke resetChannelMetadataAfterRecovery - this method mutates underlying state, so - i'm wondering if the entire method content should be synchronized?

Reply:
that's true, I put the channel assignment into the synchronized sections, so all properties are being modified there